src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +156,53 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // We always run when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            Console.WriteLine($"##vso[task.logissue type=error]Unexpected merge frequency specified ({frequency}).");
+
+            return true;
+        }
+
+        var runDateTime = DateTime.Now;
```

Comment:
Should we make sure to get this in UTC always (in case the machine time changes)?  the pipeline cron always uses UTC - https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#scheduled-triggers